jiant/models.py

```diff
@@ -687,7 +690,33 @@ def build_single_sentence_module(task, d_inp: int, project_before_pooling: bool,
     return module
 
 
-def build_pair_sentence_module(task, d_inp, model, params):
+def build_sop(task, d_inp, model, params):
+    """
+    Build and load the pretrained head for the sentence order prediction task.
+    Parameters
+    ----------
+    task: Task,
+    d_inp: int,
+    model: MultiTaskModel,
+    params: Params
+
+    Returns
+    -------
+    module: Pooler object, which is loaded with pretrained weights from ALBERT SOP
+    pretraining. 
+
+    """
+    # By default, we do not project before pooling. Since the projection weights is what is
+    # exposed in huggingface for the pretrained weights, we set projection to true.
+    project_before_pooling = True
+    module = build_pair_sentence_module(task, d_inp, model, params, project_before_pooling)
+    # The huggingface implementation exposes the pretrained projection layer for the SOP task, which
+    # we use. See: https://github.com/huggingface/transformers/issues/2671 for more details.
+    module.pooler.project = model.sent_encoder._text_field_embedder.model.pooler
+    return module
+
+
+def build_pair_sentence_module(task, d_inp, model, params, project_before_pooling=None):
```

Comment:
`project_before_pooling=None` appears to be new. Why?

Reply:
Similar questions for the changes elsewhere in this fn.

Reply:
This is because I wanted to be able to set project_before_pooling=True for this experiment (see comment in previous thread about why), but here it was set to False for transformer based models It seems as though you were the one who added this -why was this added? https://github.com/nyu-mll/jiant/blob/master/jiant/models.py#L839

Reply:
There's no reason to project (with a linear layer) _before_ pooling with BERT-style models, where pooling is just a first- or last-token selector. This kind of projection only really makes sense with max-pooling.